Framework/Algorithms/src/ConvertUnits.cpp

```diff
@@ -333,7 +340,7 @@ ConvertUnits::convertQuickly(API::MatrixWorkspace_const_sptr inputWS,
 */
 bool ConvertUnits::getDetectorValues(
     const Kernel::Unit &outputUnit, const Geometry::IComponent &source,
-    const Geometry::IComponent &sample, double l1, int emode,
+    const Geometry::IComponent &sample, double &l1, int &emode,
```

Comment:
Why non-`const` reference? In any case, I would prefer passing things like `double` and `int` by value, after all an `int` is smaller than a pointer, and you avoid some pointer overhead.